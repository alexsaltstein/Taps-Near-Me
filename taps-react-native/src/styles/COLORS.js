const COLORS = {
  white: '#ffffff',
  black: '#000000',
  transparent: '#00000000',
  green: '#18B126',
  lightpurple: '#A36BD3',
  purple: '#8250df',
  darkpurple: '#',
  gray: '#707070',
  lightgray: '#bbbbbb',
  red: '#E24242',
  blue: '#3490dc',
  gold: '#DAA520',
  transgray: '#bbbbbb40',
  transWhite: '#FFFFFF20'
}

export { COLORS }