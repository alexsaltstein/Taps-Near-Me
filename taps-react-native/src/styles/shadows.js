import { COLORS } from './COLORS';

const SHADOWS = {
  container: {
    shadowColor: COLORS.black,
    shadowOffset: {
      width: 0,
      height: 5,
    },
    shadowOpacity: 0.15,
    shadowRadius: 20,
    elevation: 5,
  },

  button: {
    shadowColor: COLORS.black,
    shadowOffset: {
      width: 0,
      height: 3,
    },
    shadowOpacity: 0.25,
    shadowRadius: 2,
    elevation: 5,
  }
}


export {SHADOWS};