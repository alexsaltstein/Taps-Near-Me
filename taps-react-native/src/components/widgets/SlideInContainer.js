import React from 'react';
import { Animated, Easing, Dimensions, StyleSheet } from 'react-native';
import { COLORS } from '../../styles/COLORS';
import { SHADOWS } from '../../styles/shadows';

const SlideInContainer = ({ children }) => {
  const slideAnim = React.useRef(new Animated.Value(0)).current;

  const doSlide = () => {
    Animated.sequence([
      Animated.timing(slideAnim, {
        toValue: 0,
        duration: 1000,
        easing: Easing.elastic(0.92),
        useNativeDriver: true,
      }),
    ]).start();
  };

  React.useEffect(() => {
    slideAnim.setValue(Dimensions.get('window').height / 2 + 100);
    doSlide();
  }, [])

  return (
    <Animated.View
      style={[{
        transform: [{ translateY: slideAnim }],
      },
      styles.container]}
    >
      {children}
    </Animated.View>
  );
};

const styles = StyleSheet.create({
  container: {
    ...SHADOWS.container,
    backgroundColor: COLORS.white,
    width: '75%',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 15,
  }
});

export default SlideInContainer;