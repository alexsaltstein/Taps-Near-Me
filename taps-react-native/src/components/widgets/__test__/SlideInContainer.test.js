import '../../../matchMedia.mock';
import React from 'react';
import renderer, { act } from 'react-test-renderer';
import SlideInContainer from '../SlideInContainer';
import {Linking, StyleSheet, Text, TouchableWithoutFeedback} from 'react-native';
import { render, fireEvent } from '@testing-library/react-native'
import ReactDOM from 'react-dom';
import { COLORS } from '../../../styles/COLORS';

const mockNavigation = {
    navigate: jest.fn(),
    addListener: jest.fn(), 
}

test('renders correctly', () => {
    const tree = renderer.create(<SlideInContainer/>).toJSON();
    expect(tree).toMatchSnapshot();
});