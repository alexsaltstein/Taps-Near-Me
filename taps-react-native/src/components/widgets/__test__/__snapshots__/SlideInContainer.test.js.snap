// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitAlignItems": "center",
      "WebkitBoxAlign": "center",
      "WebkitBoxPack": "center",
      "WebkitJustifyContent": "center",
      "WebkitTransform": "translateY(0px)",
      "alignItems": "center",
      "backgroundColor": "rgba(255,255,255,1.00)",
      "borderBottomLeftRadius": "15px",
      "borderBottomRightRadius": "15px",
      "borderTopLeftRadius": "15px",
      "borderTopRightRadius": "15px",
      "boxShadow": "0px 5px 20px rgba(0,0,0,0.15)",
      "justifyContent": "center",
      "msFlexAlign": "center",
      "msFlexPack": "center",
      "transform": "translateY(0px)",
      "width": "75%",
    }
  }
/>
`;
